#!/usr/bin/env python
PACKAGE = "aquashoko_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("roll_kp", double_t, 0, "Roll ctl PID P gain", 0.5, 0, 100)
gen.add("roll_ki", double_t, 0, "Roll ctl PID I gain", 0.1, 0, 100)
gen.add("roll_kd", double_t, 0, "Roll ctl PID D gain", 0.0, 0, 100)
gen.add("roll_mv_tf", double_t, 0, "Time constant of roll PT1 measurement filter", 0.1, 0, 10)
gen.add("roll_sp_tf", double_t, 0, "Time constant of roll PT1 reference filter", 0.1, 0, 10)


gen.add("pitch_kp", double_t, 0, "Pitch ctl PID P gain", 0.5,  0, 100)
gen.add("pitch_ki", double_t, 0, "Pitch ctl PID I gain", 0.1, 0,   100)
gen.add("pitch_kd", double_t, 0, "Pitch ctl PID D gain", 0.0, 0, 100)
gen.add("pitch_mv_tf", double_t, 0, "Time constant of pitch PT1 measurement filter", 0.1, 0, 10)
gen.add("pitch_sp_tf", double_t, 0, "Time constant of pitch PT1 reference filter", 0.1, 0, 10)

exit(gen.generate(PACKAGE, "aquashoko_control", "AttitudeControlParams"))
